File: yggdrasil/examples/transformed_io/src/transformed_io_modelC.c
#include <stdio.h>
#include <math.h>

int modelC_function(float in_val, float* in_val_copy, float* out_val) {
  in_val_copy[0] = in_val;
  out_val[0] = 2 * in_val;
  printf("modelC_function(%f) = %f\n", in_val, *out_val);
  return 0;
}

File: yggdrasil/examples/transformed_io/src/transformed_io_modelC.f90
subroutine modelC_function(in_val, in_val_copy, out_val)
  real(kind=4), intent(in) :: in_val
  real(kind=4), intent(out) :: in_val_copy
  real(kind=4), intent(out) :: out_val
  in_val_copy = in_val
  out_val = 2 * in_val
  write(*, '("modelC_function(",F10.5,") = ",F10.5)') in_val, out_val
end subroutine modelC_function
